pkg/chunk/chunk_store_utils.go

```diff
@@ -252,3 +256,10 @@ func (c *Fetcher) processCacheResponse(chunks []Chunk, keys []string, bufs [][]b
 	}
 	return found, missing, err
 }
+
+func injectShardLabels(chunks []Chunk, shard astmapper.ShardAnnotation) {
```

Comment:
I agree, but put it here in order to avoid a circular dependency between `astmapper` and `chunk` pkgs. I can investigate a way to do this via injection, a sidecar package, etc, but I felt this would be cumbersome and another layer of confusion in an already large PR. Let me know what you'd prefer.

Reply:
It's fine to keep it here for now.